lm_eval/tasks/codexglue_code_refinement.py

```diff
@@ -0,0 +1,136 @@
+journal = {ACM Trans. Softw. Eng. Methodol.},
+month = {sep},
+articleno = {19},
+numpages = {29},
+keywords = {Neural machine translation, bug-fixes}
+}
+"""
+
+FUNCTION_LENGTH = {
+    "small": "small",
+    "medium": "medium",
+}
+
+
+def create_all_tasks():
+    """Creates a dictionary of tasks from a list of sizes based on the function length
+    :return: {task_name: task}
+        e.g. {codexglue_code_refinement-small: Task, codexglue_code_refinement-medium: Task}
+    """
+    return {
+        f"codexglue_code_refinement-{func_length}": create_task(func_length)
+        for func_length in FUNCTION_LENGTH
+    }
+
+
+def create_task(func_length):
+    class CodexGLUECodeRefinementTask(CodexGLUECodeRefinement):
+        def __init__(self):
+            super().__init__(func_length)
+
+    return CodexGLUECodeRefinementTask
+
+
+class CodexGLUECodeRefinement(Task):
+    """A task represents an entire benchmark including its dataset, problems,
+    answers, generation settings and evaluation methods.
+    """
+
+    DATASET_PATH = "code_x_glue_cc_code_refinement"
+    DATASET_NAME = None
+
+    def __init__(self, func_length):
+        self.DATASET_NAME = func_length
+        super().__init__(
+            stop_words=["\n"],
+            requires_execution=False,
+        )
+
+    def get_dataset(self):
+        """Returns dataset for the task or an iterable of any object, that get_prompt can handle"""
+        return self.dataset["test"]
+
+    def fewshot_examples(self):
+        """Loads and returns the few-shot examples for the task if they exist."""
+        with open(
+            "lm_eval/tasks/few_shot_examples/codexglue_code_refinement_few_shot_prompts.json",
+            "r",
+        ) as file:
+            examples = json.load(file)
+        return examples
+
+    @staticmethod
+    def two_shot_prompt(entry, text, examples):
+        """Two shot prompt format as Buggy Version of Java Code & Fixed Version of Java Code"""
+        prompt = f"\nBuggy Version of Java Code:\n{examples['buggy version of java code1']}\
+                   \nFixed Version of Java Code:\n{examples['fixed version of java code1']}\
+                   \nBuggy Version of Java Code:\n{examples['buggy version of java code2']}\
+                   \nFixed Version of Java Code:\n{examples['fixed version of java code2']}\
+                   \nBuggy Version of Java Code:\n{text}\
+                   \nFixed Version of Java Code:\n"
+        return entry + prompt
+
+    def get_prompt(self, doc):
+        """Builds the prompt for the LM to generate from."""
+        func_length = FUNCTION_LENGTH[self.DATASET_NAME]
+        text = doc["buggy"]
+        entry = f"Convert this {func_length} sized buggy version of Java function's code into it's correct version:\n"
```

Comment:
see comment above about changing intro phrase to `Fix this  buggy Java function:\n `